odom_frame: odom
base_link_frame: base_link
world_frame: odom

two_d_mode: true

frequency: 50

# odom0: smb_diff_drive/odom
# # x, y, z, roll, pitch, yaw, vx, vy, vz, vroll, vpitch, vyaw, ax, ay, az
# odom0_config: [false, false, false,
#                false, false, false,
#                true, true, true,
#                false, false, true,
#                false, false, false]
# odom0_queue_size: 100
# odom0_differential: false
# odom0_relative: false

# odom1: tracking_camera/odom/sample
# # x, y, z, roll, pitch, yaw, vx, vy, vz, vroll, vpitch, vyaw, ax, ay, az
# odom1_config: [false, false, false,
#                false, false, false,
#                true, true, true,
#                false, false, true,
#                false, false, false]
# odom1_queue_size: 100
# odom1_differential: false
# odom1_relative: false

# imu0: imu
# imu0_config: [false, false, false,
#               true, true, true,
#               false, false, false,
#               true, true, true,
#               false, false, false]
# imu0_differential: true
# imu0_queue_size: 10
# imu0_remove_gravitational_acceleration: true

odom0: /tracking_camera/odom/sample
odom0_config: [false, false, false,
               false, false, false,
               true, true, true,
               false, false, true,
               false, false, false]
odom0_queue_size: 100
odom0_differential: false
odom0_relative: false

# twist0: /tracking_camera/odom/sample_transformed
# twist0_config: [false, false, false,
#                 false, false, false,
#                 true,  true,  true,
#                 false, false, false,
#                 false, false, false]
# twist0_queue_size: 3
# twist0_rejection_threshold: 2
# twist0_nodelay: false

publish_tf: true

# Diagnostic configuration
diagnostic_updater:
  frequency: 1.0   # Frequency in Hz for publishing diagnostics
  min_frequency: 0.1
  max_frequency: 100.0

print_diagnostics: true
